src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -115,7 +115,7 @@ function navigateToFromDotNet(uri: string, options: NavigationOptions): void {
 function navigateToCore(uri: string, options: NavigationOptions, skipLocationChangingCallback = false): void {
   const absoluteUri = toAbsoluteUri(uri);
 
-  if (!options.forceLoad && isWithinBaseUriSpace(absoluteUri)) {
+  if (!options.forceLoad && isWithinBaseUriSpace(absoluteUri) && hasInteractiveRouter()) {
```

Comment:
I suspect this might not be a sufficient solution anyway because if it uses `performExternalNavigation` then it won't do enhanced navigation. We would still want enhanced navigation in this case. You can trigger that by calling `performEnhancedPageLoad`.